pyEpiabm/pyEpiabm/sweep/initial_place_sweep.py

```diff
@@ -0,0 +1,127 @@
+                                              person_weights=weights,
+                                              mean_capacity=mean_cap)
+
+                elif place.place_type.name == "Workplace":  # WORKSPACE
+                    # Fixed population is initialised on first run
+                    # thinking of making sure people don't have more than one
+                    # workplace
+                    power_list = [max_size, offset, power]
+                    helper.update_place_group(place, group_size=ave_group_size,
+                                              person_list=person_list,
+                                              person_weights=weights,
+                                              mean_capacity=mean_cap,
+                                              power_law_params=power_list)
+
+                elif place.place_type.name == "CareHome":  # CAREHOME
+                    # Kit will add more detail for Carehomes
+                    helper.update_place_group(place)
+
+        # Instantiate the temporary population in each place using
+        # the update sweep.
+        add_temporary_population = UpdatePlaceSweep()
+        add_temporary_population.bind_population(self._population)
+        add_temporary_population(0)
+
+    def create_age_weights(self, place, params):
+        """Function to return a list of people in the correct
+        age range to be added to a place, and the weights
+        for the different age groups.
+
+        Parameters
+        ----------
+
+        place : Place
+            Place to add people to
+        params : dict
+            Dictionary of parameters with age structure data
+
+        Returns
+        -------
+
+        typing.List[Person]
+            List of people who may be in the place
+        typing.List[float]
+            Corresponding weights for the person list
+
+        """
+        param_ind = place.place_type.value - 1
+        min_age = [params["age_group1_min_age"][param_ind],
+                   params["age_group2_min_age"][param_ind],
+                   params["age_group3_min_age"][param_ind]]
+        max_age = [params["age_group1_max_age"][param_ind],
+                   params["age_group2_max_age"][param_ind],
+                   params["age_group3_max_age"][param_ind]]
+        prop = [params["age_group1_prop"][param_ind],
+                params["age_group2_prop"][param_ind],
+                params["age_group3_prop"][param_ind]]
+
+        person_list = []
+        weights = []
+        for person in place.cell.persons:
+            if (place.place_type in person.place_types):
+                # People can't have more than one place of each type.
+                continue
+            for i in range(3):
+                if (person.age > (min_age[i]-1) and person.age < max_age[i]):
+                    # Assumes age groups are distinct and integers.
+                    person_list.append(person)
+                    weights.append(prop[i])
```

Comment:
Is it worth breaking at the end of this, to make sure that people can't be added to multiple places (even if the age groups do overlap)? Or actually do we want this to be possible?

Reply:
Think break is a good idea as stops people being added to the same place twice if they are boarderline on age